### DISCUSSION TITLE
How to ignore old messages sent when the bot was offline

### DISCUSSION THREAD

**Discussion by Abo5** [atipugin/telegram-bot-ruby#304] [Q&A] [answered]
Hi,
Let's say I have a very simple bot
```
require 'telegram/bot'

Telegram::Bot::Client.run(ENV['TOKEN']) do |bot|
  bot.listen do |message|
    case message.text
    when '/start'
      bot.api.send_message(chat_id: message.chat.id, text: 'Hello!')
    when '/stop'
      bot.api.send_message(chat_id: message.chat.id, text: 'Goodbye!')
    when 'Hi'
      bot.api.send_message(chat_id: message.chat.id, text: 'Hello')
    end
  end
end
```

And someone spammed Hi in the group chat while the bot was offline, how can I make the bot ignore all the messages that was sent when it was offline? Knowing that there's no database to store when it came back alive.

**Comment by AlexWayfer** [comment, marked as answer]
I guess, it's not a question to the library. It's up to your application (bot), business logic.

Commonly, you can process these messages and response as regular (but with delay).

But… to ignore them — you have to keep some marker in application, like a start time, and compare incoming messages time with it, for example.

As I know, Telegram will forget about these pending messages after some time, maybe around 1 hour, or 24 hours.

**Comment by wteuber** [comment]
Checking the time works well for me, no need for custom markers...
Only use this pattern if you a willing to ignore all messages sent to the bot between shutdown and restart.

```ruby
require 'telegram/bot'

start_time = Time.now.utc

Telegram::Bot::Client.run(ENV['TOKEN']) do |bot|
  bot.listen do |message|

    next unless message.date >= start_time.to_i

    case message.text
    when '/start'
      bot.api.send_message(chat_id: message.chat.id, text: 'Hello!')
    when '/stop'
      bot.api.send_message(chat_id: message.chat.id, text: 'Goodbye!')
      exit
    when 'Hi'
      bot.api.send_message(chat_id: message.chat.id, text: 'Hello')
    end
  end
end
```
